Realm/Realm/DataBinding/WovenSetterMethodInfo.cs

```diff
@@ -25,54 +25,69 @@ namespace Realms.DataBinding
+        public override Type DeclaringType => _setterMi.DeclaringType;
 
-        public override string Name => _mi.Name;
+        public override RuntimeMethodHandle MethodHandle => _setterMi.MethodHandle;
 
-        public override Type ReflectedType => _mi.ReflectedType;
+        public override string Name => _setterMi.Name;
 
-        public override ICustomAttributeProvider ReturnTypeCustomAttributes => _mi.ReturnTypeCustomAttributes;
+        public override Type ReflectedType => _setterMi.ReflectedType;
 
-        public override Type ReturnType => _mi.ReturnType;
+        public override ICustomAttributeProvider ReturnTypeCustomAttributes => _setterMi.ReturnTypeCustomAttributes;
 
-        public WovenSetterMethodInfo(MethodInfo mi)
+        public override Type ReturnType => _setterMi.ReturnType;
+
+        public WovenSetterMethodInfo(MethodInfo setterMi, MethodInfo getterMi)
         {
-            Argument.NotNull(mi, nameof(mi));
-            _mi = mi;
+            Argument.NotNull(setterMi, nameof(setterMi));
+            Argument.NotNull(getterMi, nameof(getterMi));
+
+            _setterMi = setterMi;
+            _getterMi = getterMi;
         }
 
-        public override MethodInfo GetBaseDefinition() => _mi.GetBaseDefinition();
+        public override MethodInfo GetBaseDefinition() => _setterMi.GetBaseDefinition();
 
-        public override object[] GetCustomAttributes(bool inherit) => _mi.GetCustomAttributes(inherit);
+        public override object[] GetCustomAttributes(bool inherit) => _setterMi.GetCustomAttributes(inherit);
 
-        public override object[] GetCustomAttributes(Type attributeType, bool inherit) => _mi.GetCustomAttributes(attributeType, inherit);
+        public override object[] GetCustomAttributes(Type attributeType, bool inherit) => _setterMi.GetCustomAttributes(attributeType, inherit);
 
-        public override MethodImplAttributes GetMethodImplementationFlags() => _mi.GetMethodImplementationFlags();
+        public override MethodImplAttributes GetMethodImplementationFlags() => _setterMi.GetMethodImplementationFlags();
 
-        public override ParameterInfo[] GetParameters() => _mi.GetParameters();
+        public override ParameterInfo[] GetParameters() => _setterMi.GetParameters();
 
         public override object Invoke(object obj, BindingFlags invokeAttr, Binder binder, object[] parameters, CultureInfo culture)
         {
-            var managingRealm = (obj as IRealmObjectBase)?.Realm;
+            var realmObject = obj as IRealmObjectBase;
 
-            // If managingRealm is not null and not currently in transaction, wrap setting the property in a realm.Write(...)
-            if (managingRealm?.IsInTransaction == false)
+            if (realmObject?.IsManaged == true)
             {
-                return managingRealm.Write(() =>
+                if (_getterMi.Invoke(realmObject, null).Equals(parameters[0]))
```

Comment:
I'm not crazy about the reflection here. I would've preferred punching straight to the accessor's GetValue. But this is by definition not a hot path, so I suppose it doesn't really matter. But a comment explaining why we do this is necessary, ideally linking to the issue for the full context.

Reply:
I'll definitely add a comment. 
I was thinking about using `GetValue` here, but it starts to get a little more complex because we also don't know exactly what is the `mapTo` name of the property